src/Umbraco.PublishedCache.HybridCache/Services/DocumentCacheService.cs

```diff
@@ -101,6 +109,20 @@ public DocumentCacheService(
             {
                 using ICoreScope scope = _scopeProvider.CreateCoreScope();
                 ContentCacheNode? contentCacheNode = await _databaseCacheRepository.GetContentSourceAsync(key, preview);
+
+                // If we can resolve the content cache node, we still need to check if the ancestor path is published.
```

Comment:
I'm sure you are right here, but I just want to double-check that we need this (and understand why we do).   As I was thinking so long as we make sure all descendants are removed from the cache when an ancestor is unpublished, we should never get into a state where we find a node in the cache under an unpublished ancestor. And hence we wouldn't need to do the check here.

Reply:
It's a bit weird here, but we need this check because the "Umbraco cache" actually also has 2 levels: the in-memory cache and the database cache. The purpose of this is to quickly build the in-memory cache. But the database cache always has **all** published nodes, regardless of whether their ancestors are published or not, so that's why.

But now that I think of it, I think we could optimize this by asking for the published ancestor path **before** looking for the node in the database, that way we skip a database lookup if one of the ancestors are unpublished anyways. I'll go ahead and make this change :smile: